src/server/game/AI/CreatureAI.h

```diff
@@ -152,15 +150,14 @@ class TC_GAME_API CreatureAI : public UnitAI
         virtual void ReceiveEmote(Player* /*player*/, uint32 /*emoteId*/) { }
 
         // Called when owner takes damage
-        virtual void OwnerAttackedBy(Unit* /*attacker*/) { }
+        virtual void OwnerAttackedBy(Unit* attacker);
 
         // Called when owner attacks something
-        virtual void OwnerAttacked(Unit* /*target*/) { }
+        virtual void OwnerAttacked(Unit* target) { CreatureAI::OwnerAttackedBy(target); }
```

Comment:
You talked about traps for scripters but here you add another one - it would be better to introduce a separate (nonvirtual) method that is called by both OwnerAttacked and OwnerAttackedBy. Someone less experienced might think that if he overrides OwnerAttackedBy in his own script but not OwnerAttacked that it will call his overriden OwnerAttackedBy (but it will be the base CreatureAI one)